build.rs

```diff
@@ -13,4 +13,19 @@ fn main() {
     {
         println!("cargo::rustc-cfg=feature=\"include-ui\"");
     }
+    // Create a new, empty icon collection
+    let mut icon_dir = ico::IconDir::new(ico::ResourceType::Icon);
+    // Read a PNG file from disk and add it to the collection
+    let file = std::fs::File::open("assets/icon.png").expect("Could not find assets/icon.png");
```

Comment:
I think if there's no icon file it should still build but just not embed the icon, but  other errors should still cause it to fail. You would also need to check the specific reason why it failed in that case.

I think something along these lines should work:
```rust
match std::fs::File::open("assets/icon.png") {
    Ok(file) => {
        // Embedding stuff in here
    },
    Err(e) if e.kind() == std::io::ErrorKind::NotFound => {},
    Err(e) => {
        panic!("Couldn't open icon file: {e}");
    }
}
```

Reply:
Functions as intended. I just put an underscore before 'file' so clippy is happy.